runtime/lua/vim/fs.lua

```diff
@@ -679,4 +679,22 @@ function M.rm(path, opts)
   end
 end
 
+--- Returns `true` if `path` is relative to `root`
+--- @param path string?
+--- @param root string?
+--- @return boolean
+function M.is_relative_to(path, root)
+  if root == nil or path == nil then
+    return false
+  end
+  path = vim.fs.normalize(path)
+  root = vim.fs.normalize(root)
+  for dir in vim.fs.parents(path) do
```

Comment:
did we decide not to use the  absolute path or `realpath`  approach? avoids an iterator.

Reply:
No decision has been made as far as I'm aware. I went with this approach mostly because it seemed safer, as to avoid false positives for edge cases where `root` is `/home/user` and `path` is `/home/username`. If that's not really a concern, then the following code snippet should work I think:

```lua
function M.is_relative_to(path, root)
  vim.validate('path', path, 'string')
  vim.validate('root', root, 'string')
  return root == path or vim.startswith(vim.fs.normalize(path), vim.fs.normalize(root) .. '/')
end
```

Shall we go with this?

Reply:
Answer: we shall not, because it fails the tests. So we could go with absolute/realpath but we'd need to ensure it passes the tests. I welcome any challenger :)

Reply:
No decision has been made, but I'm not sure it can be done in a robust way. I tried but it didn't pass the tests.